lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -303,10 +360,16 @@ export class NativeInteractionClient extends BaseInteractionClient {
             this.validateNativeResponse(response);
         } catch (e) {
             // Only throw fatal errors here to allow application to fallback to regular redirect. Otherwise proceed and the error will be thrown in handleRedirectPromise
-            if (e instanceof NativeAuthError && isFatalNativeAuthError(e)) {
-                throw e;
+            if (e instanceof NativeAuthError) {
+                this.serverTelemetryManager.setNativeBrokerErrorCode(
+                    e.errorCode
+                );
+                if (isFatalNativeAuthError(e)) {
+                    throw e;
+                }
             }
         }
+        this.serverTelemetryManager.clearNativeBrokerErrorCode();
```

Comment:
Why do we set the error code and then immediately clear it?

Reply:
Sorry, this was supposed to be added to `handleRedirectPromise`. Good catch, thank you. Fixed. Added a few unit tests to cover this scenario